zerver/actions/message_send.py

```diff
@@ -265,6 +283,26 @@ def notification_recipients(setting: str) -> Set[int]:
             # by default for new users.)
             wildcard_mention_user_ids = notification_recipients("wildcard_mentions_notify")
 
+        if possible_wildcard_mention and possible_topic_wildcard_mention:
+            # If the possibility exists for both the wildcard and topic wildcard mentions,
+            # we need to separately store the corresponding users to be notified.
+            #
+            # We can't give preference to wildcard mention and store
+            # 'wildcard_mention_user_ids' only, because after rendering it might be
+            # possible that there isn't any wildcard mention involved, only topic wildcard mention,
+            # e.g., @all in the code block and genuine @topic mention.
+            for user_id in wildcard_mention_user_ids:
+                if user_id in possibly_topic_wildcard_mentioned_user_ids:
+                    topic_wildcard_mention_user_ids.add(user_id)
+
+        if not possible_wildcard_mention and possible_topic_wildcard_mention:
+            possible_topic_wildcard_mention_user_ids = notification_recipients(
+                "wildcard_mentions_notify"
+            )
```

Comment:
It would be better to calculate the set of people who have enabled the setting earlier inside an `possible_wildcard_mention OR possible_topic_wildcard_mention` conditional (and call this set simply as `wildcard_mentions_enabled_user_ids` or something), and avoid the super-confusing possible/possibly naming all-together.

Reply:
Avoided the possible/possibly naming and [re-wrote these conditionals](https://github.com/zulip/zulip/pull/25828/commits/514f0def1ad9640fa3ac6baeb8010b314e7bbb17#diff-d803e860e5ff2068cb51aab642015247be71eb4045444a4dcb82e86486f22a4dR276) in a much simpler and better way.